rpc/handlers.go

```diff
@@ -775,9 +779,80 @@ func getAddInvokeTxCode(err error) int {
 		return jsonrpc.InvalidParams
 	case strings.Contains(err.Error(), "not supported. Supported versions"):
 		return jsonrpc.InvalidParams
-	case strings.Contains(err.Error(), "max_fee must be bigger than 0.\n0 >= "):
+	case strings.Contains(err.Error(), "max_fee must be bigger than 0"):
 		return jsonrpc.InvalidParams
 	default:
 		return jsonrpc.InternalError
 	}
 }
+
+// AddDeclareTransaction relays a declare transaction to the gateway.
+//
+// It follows the specification defined here:
+// https://github.com/starkware-libs/starknet-specs/blob/a789ccc3432c57777beceaa53a34a7ae2f25fda0/api/starknet_write_api.json#L39
+// Note: The gateway expects the sierra_program to be gzip compressed and 64-base encoded. We perform this operation,
+// and then relay the transaction to the gateway.
+func (h *Handler) AddDeclareTransaction(declareTx *json.RawMessage) (*DeclareTxResponse, *jsonrpc.Error) {
```

Comment:
```suggestion
func (h *Handler) AddDeclareTransaction(declareTx map[string]any) (*DeclareTxResponse, *jsonrpc.Error) {
```

Let the jsonrpc package do the unmarshaling for you

Reply:
might make testing harder tho, so up to you

Reply:
Thanks for the suggestion. I'm happy to leave it as is because` AddTransaction()` takes` json.RawMessage`, and if I accepted a `map[string]any`, I think I would have to cast it to `json.RawMessage` anyway to use in `AddTransaction()`